src/docs/development/testing.mdx

```diff
@@ -38,24 +80,90 @@ yarn test
+
+Functional tests alone will not give you enough test coverage to avoid regressions. Unit tests aim to provide code coverage depth, while functional tests provide coverage over the breadth of user test scenarios.
+
+## How to Write
 
-End-to-End tests are a higher level test than Unit Tests. They focus primarily on the flow of the application, making sure that the different parts of the application work together correctly. For example, you may test that clicking a login button takes you to the appropriate sign-in page.
+End-to-end tests rely on the ability to find elements on the screen and interact with them in an automated way. This means that you need a reliable mechanism for selecting elements. Depending on your testing framework, there are different ways to select elements, including:
+
+-   **CSS Selectors**: find elements by using CSS classes or rules (e.g., `.your-class-name`)
+-   **Element Selectors**: find elements inside the elements (e.g., `#\@\@pxb-drawerlayout-content > div > header > div > button`)
+-   **xPath Selectors**: find selector within xPath (e.g., `//*[@id="@@pxb-drawerlayout-content"]/div/header/div/button`)
+-   **ID Selectors**: find attribute id within elements (e.g., `#login-button`)
+
+See selector strategies on [browserstack](https://www.browserstack.com/guide/css-selectors-in-selenium) and [software testing help](https://www.softwaretestinghelp.com/css-selector-selenium-locator-selenium-tutorial-6/)
+
+Each of these methods has their own advantages and disadvantages. A common pain point in E2E test automation is that modern Javascript platforms are constantly changing, particularly in the open source community. Many of these frameworks automatically (and dynamically) generate IDs and classes for components resulting in a constantly moving target for automation tooling. Identifying elements by these values (i.e., using IDs, CSS, or xPath locators _without_ property attributes) makes your tests brittle because they are subject to change any time a new version comes out (or even any time the page is reloaded).
+
+One way to combat this in your own applications is to use dedicated attributes for testing (e.g., `test-id` or `dev-id`). By adding this test-specific attribute to the element, identifying the correct underlying component should be safer. Because this attribute is test-specific and the application logic does not use it, it will be less likely to change throughout the product lifecycle. Combining this approach with other selectors is a common and reliable approach for element identification.
+
+```html
+<button test-id="login-button" class="default-button">Login</button>
+```
+
+Selector to use in a test
+
+```
+WebElement click = driver.findElement(By.id("login-button"));
+```
```

Comment:
Does by.id work here if the attribute is test-id?

Reply:
```suggestion
```html
// Component code
<button test-id="login-button" class="default-button">Login</button>

// Selector to use in a test
WebElement click = driver.findElement(By.id("login-button"));
```
```